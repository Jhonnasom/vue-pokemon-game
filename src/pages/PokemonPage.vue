<template>
  <h1 v-if="!pokemon">Espere por favor...</h1>
  <div v-else>
    <h1>Â¿Quien es este Pokemon?</h1>

    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
    <PokemonOptions :pokemons="pokemonArr" @selection="checkAnswer" />

    <template v-if="showAnswer">
      <h2 class="fade-in">{{ message }}</h2>
      <button @click="newGame">Play Again</button>
    </template>
  </div>
</template>

<script>
import PokemonOptions from "../components/PokemonOptions.vue";
import PokemonPicture from "../components/PokemonPicture.vue";
import getPokemonOptions from "../helpers/getPokemonOptions";

// console.log(getPokemonOptions());

export default {
  name: "PokemonPage",
  components: {
    PokemonOptions,
    PokemonPicture,
  },
  data() {
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false,
      showAnswer: false,
      message: "",
    };
  },
  methods: {
    async mixPokemonsArr() {
      this.pokemonArr = await getPokemonOptions();

      // Floor es para redondear y se multiplica por 4 xq quiero que este # este entre 0 y 3
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonArr[rndInt];
    },
    checkAnswer(pokemonId) {
      this.showPokemon = true;
      this.showAnswer = true;

      if (pokemonId === this.pokemon.id) {
        this.message = "Correcto, es " + this.pokemon.name + "!";
      } else {
        this.message = "Ooops, era " + this.pokemon.name + "!";
      }
      console.log("Hola mundo", this.showPokemon);

      // if (pokemonId === this.pokemon.id) {
      //   this.showPokemon = true;
      //   this.message = "Correcto, es " + this.pokemon.name + "!";
      //   console.log("Resolviste correctamente");
      // } else {
      //   console.log("Pokemon incorrecto");
      //   this.message = "Ooops, era " + this.pokemon.name + "!";
      //   this.showPokemon = false;
      // }
      // setTimeout(() => {
      //   this.showPokemon = false;
      //   this.mixPokemonsArr();
      // }, 2000);
    },
    newGame() {
      this.showPokemon = false;
      this.showAnswer = false;
      this.pokemon = null;
      this.mixPokemonsArr();
    },
  },
  mounted() {
    this.mixPokemonsArr();
  },
};
</script>

<style></style>
<!--  -->
