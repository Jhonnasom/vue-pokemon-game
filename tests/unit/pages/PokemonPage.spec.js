import { shallowMount,mount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonPage Component', () => { 
  let wrapper

  beforeEach(() => { 
    wrapper = shallowMount(PokemonPage)
  })
  
  test('debe de hacer match con el snaptshot ', () => { 
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de llamar mixPokemonArray al montar', () => {
    
    const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
    const wrapper = shallowMount(PokemonPage)

    expect(mixPokemonArraySpy).toHaveBeenCalled()
  })
  
  test('debe de hacer match con el snaptshot cuando cargan los pokemons', () => {
      
      const wrapper = mount(PokemonPage, { 
        data() { 
          return { 
            pokemonArr: pokemons,
            pokemon: pokemons[0],
            showPokemon: false,
            showAnswer: false,
            message: ''
          }
        }
      })
  
      expect(wrapper.html()).toMatchSnapshot()
  })
  test('debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => {
    const wrapper = shallowMount(PokemonPage, { 
      data() { 
        return { 
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: ''
        }
      }
    })

    const picture = wrapper.find('pokemon-picture-stub')
    const options = wrapper.find('pokemon-options-stub')
    // PokemonPicture debe de existir
     expect(picture.exists()).toBeTruthy()
    //  PokemonOptions deben de existir
    expect(options.exists()).toBeTruthy()

    // PokemonPicture attribute pokemonId===5
    expect(picture.attributes('pokemonid')).toBe('5')

    // PokemonOptions attribute pokemons to be true
    expect(options.attributes('pokemons')).toBeTruthy()

   })
  
 })