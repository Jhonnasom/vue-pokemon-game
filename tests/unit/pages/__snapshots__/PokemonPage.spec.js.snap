// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe de hacer match con el snaptshot  1`] = `<h1>Espere por favor...</h1>`;

exports[`PokemonPage Component debe de hacer match con el snaptshot cuando cargan los pokemons 1`] = `
<div>
  <h1>Â¿Quien es este Pokemon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon"></div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
